azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -73,11 +92,56 @@ def from_json(cls, json_string: str):
             New instance of the durable orchestration context class
         """
         # We should consider parsing the `Input` field here as well,
-        # intead of doing so lazily when `get_input` is called.
+        # instead of doing so lazily when `get_input` is called.
         json_dict = json.loads(json_string)
         return cls(**json_dict)
 
-    def call_activity(self, name: str, input_: Optional[Any] = None) -> Task:
+    def _generate_task(self, action: Action,
+                       retry_options: Optional[RetryOptions] = None) -> Union[AtomicTask,
```

Comment:
Just to clarify, we don't need to return `CompoundTasks` here because they are in general just composed of individual `AtomicTask`, so we don't want to increment the TaskId, right?

Reply:
That's exactly right. Just to add more context: with this new algorithm we're quite dependent (sometimes to a scary extent) in the correctness of the `TaskId`. The reason for this is that we need to keep the `TaskId` in sync with what the C# histories will produce and, in C#, the `TaskId` is not increment when `WhenAny` and `WhenAll` are called.